packages/editor/src/components/post-publish-button/index.js

```diff
@@ -80,6 +80,8 @@ export class PostPublishButton extends Component {
 				entitiesSavedStatesCallback();
 			}
 		} );
+		// Clear the callback set for entities saved states.
+		this.props.setCloseEntitiesCallback();
```

Comment:
This is the responsibility of `closeEntitiesSavedStates`.

Reply:
Agreed.  This is the last line of `closeEntitiesSavedStates`.

Reply:
I meant that the callback should be cleared here https://github.com/WordPress/gutenberg/blob/9ea1a04178dc5d736b18f363e4f78bc0535143e6/packages/edit-post/src/components/layout/index.js#L142-L150.

Not by the caller. It's something that always should happen after calling the callback. Not something that the caller should decide on.

Reply:
Ah ok, that makes sense!  

So instead of relying on `closeEntitiesCallback` to have been defined in a way that resets the callback after use, we just clear the callback after calling it.  Thanks again =) 

This is updated.